src/config.py

```diff
@@ -0,0 +1,253 @@
+import logging
+from pathlib import Path
+from typing import Annotated, Any, ClassVar, Optional
+
+import tomllib
+from dotenv import load_dotenv
+from pydantic import Field, field_validator
+from pydantic_core.core_schema import ValidationInfo
+from pydantic_settings import (
+    BaseSettings,
+    PydanticBaseSettingsSource,
+    SettingsConfigDict,
+)
+
+# Load .env file for local development.
+# Make sure this is called before AppSettings is instantiated if you rely on .env for AppSettings construction.
+load_dotenv()
+
+logger = logging.getLogger(__name__)
+
+
+def load_toml_config(config_path: str = "config.toml") -> dict[str, Any]:
+    """Load configuration from TOML file if it exists."""
+    config_file = Path(config_path)
+    if config_file.exists():
+        try:
+            with open(config_file, "rb") as f:
+                return tomllib.load(f)
+        except (tomllib.TOMLDecodeError, OSError) as exc:
+            logger.warning("Failed to load %s: %s", config_path, exc)
+            return {}
+    return {}
+
+
+# Load TOML config once
+TOML_CONFIG = load_toml_config()
+
+
+class TomlConfigSettingsSource(PydanticBaseSettingsSource):
+    """Custom settings source for loading from TOML file."""
+
+    def __init__(self, settings_cls: type[BaseSettings]) -> None:
+        super().__init__(settings_cls)
+
+    SECTION_MAP: ClassVar[dict[str, str]] = {
+        "DB": "db",
+        "AUTH": "auth",
+        "SENTRY": "sentry",
+        "OPENTELEMETRY": "opentelemetry",
+        "LLM": "llm",
+        "AGENT": "agent",
+        "DERIVER": "deriver",
+        "HISTORY": "history",
+        "": "app",  # For AppSettings with no prefix
+    }
+
+    def get_field_value(self, field: Field) -> tuple[Any, str, bool]:
```

Comment:
As far as I can see, this function is never used in the codebase. Can it be removed? Also, it gets red squiggly underlines in my IDE for "Field" being an overload rather than a class (?)